streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -918,21 +984,19 @@ void validate(final Set<String> copartitionGroup,
             for (final String topic : copartitionGroup) {
                 if (!allRepartitionTopicsNumPartitions.containsKey(topic)) {
                     final Integer partitions = metadata.partitionCountForTopic(topic);
-
                     if (partitions == null) {
-                        throw new org.apache.kafka.streams.errors.TopologyException(String.format("%sTopic not found: %s", logPrefix, topic));
+                        String str = String.format("%sTopic not found: %s", logPrefix, topic);
+                        log.error(str);
+                        throw new org.apache.kafka.streams.errors.TopologyException(str);
```

Comment:
We should not use a `TopologyException` here any more since `TopologyException` should be used for DSL statement parsing only. Instead we could just throw an IllegalStateException since it should never be expected (i.e. if the metadata is indeed not known we should error out with the error assignment earlier and never reach this line).